site/src/pages/CreateWorkspacePage/CreateWorkspacePage.tsx

```diff
@@ -23,126 +23,112 @@ import {
   templateByName,
   templateVersionExternalAuth,
 } from "api/queries/templates";
-import { autoCreateWorkspace, createWorkspace } from "api/queries/workspaces";
+import {
+  AutoCreateWorkspaceOptions,
+  autoCreateWorkspace,
+  createWorkspace,
+} from "api/queries/workspaces";
 import { checkAuthorization } from "api/queries/authCheck";
 import { CreateWSPermissions, createWorkspaceChecks } from "./permissions";
 import { richParameters } from "api/queries/templateVersions";
-import { paramsUsedToCreateWorkspace } from "utils/workspace";
 import { useEffectEvent } from "hooks/hookPolyfills";
+import { workspaceBuildParameters } from "api/queries/workspaceBuilds";
 
-type CreateWorkspaceMode = "form" | "auto";
+export const createWorkspaceModes = ["form", "auto", "duplicate"] as const;
+export type CreateWorkspaceMode = (typeof createWorkspaceModes)[number];
 
-export type ExternalAuthPollingState = "idle" | "polling" | "abandoned";
+export type ExternalAuthPollingStatus = "idle" | "polling" | "abandoned";
 
 const CreateWorkspacePage: FC = () => {
   const organizationId = useOrganizationId();
+  const [searchParams] = useSearchParams();
```

Comment:
Sorry for all the shuffling around, but I generally think it's better to keep all the hook calls at the top of the component, instead of mingling them with derived values that aren't necessary for other hook calls

Much easier to avoid rule of hooks violations